test/controllers/AllowancesControllerTest.scala

```diff
@@ -52,79 +50,97 @@ class AllowancesControllerTest extends UnitSpec with FakeTaxsPlayApplication wit
     surname = "surname"
   )
 
+  val genericViewModel: GenericViewModel =  AtsList(
+    utr = "3000024376",
+    forename = "forename",
+    surname = "surname",
+    yearList = List(
+      TaxYearEnd(Some("2015"))
+    )
+  )
+
   trait TestController extends AllowancesController {
     override lazy val allowanceService = mock[AllowanceService]
     override lazy val auditService = mock[AuditService]
     implicit lazy val formPartialRetriever: FormPartialRetriever = AppFormPartialRetriever
-
     val model = baseModel
+    implicit val request = FakeRequest("GET","?taxYear=2015")
+    implicit val user = User(AuthorityUtils.saAuthority(testOid, testUtr))
+    when(allowanceService.getAllowances(Matchers.eq(taxYear))(Matchers.eq(user),Matchers.eq(request),Matchers.any())).thenReturn(Future.successful(baseModel))
 
-    when(allowanceService.getAllowances(any[User], any[Request[AnyRef]], any[HeaderCarrier])).thenReturn(model)
   }
 
   "Calling allowances with no session" should {
 
     "return a 303 response" in new TestController {
-
       val result = Future.successful(authorisedAllowance(request))
       status(result) shouldBe 303
     }
+
   }
 
-  "Calling allowances with session" should {
+"Calling allowances with session" should {
 
-    "have the right user data in the view" in new TestController {
+  "return a Future[Either[ErrorResponse,GenericViewModel]] when extractModel is called " in new TestController {
```

Comment:
Test name does not correspond to what's being tested. You aren't testing the return type here. Tests should be able to be read and understood with little context.

Reply:
Tests Changed